workflow/executor/executor.go

```diff
@@ -724,19 +715,25 @@ func (we *WorkflowExecutor) CaptureScriptResult(ctx context.Context) error {
 // AnnotateOutputs annotation to the pod indicating all the outputs.
 func (we *WorkflowExecutor) AnnotateOutputs(ctx context.Context, logArt *wfv1.Artifact) error {
 	outputs := we.Template.Outputs.DeepCopy()
+	containerName := common.MainContainerName
 	if logArt != nil {
+		containerName = strings.TrimSuffix(logArt.Name, "-logs")
```

Comment:
this is too hacky - can we find another way?

Reply:
i moved the call to `AnnotateOutputs` into the loop on the containerNames, right after the call for `SaveLogs`, so now i send the name as a parameter.
but i can also just change `SaveLogs` to both save the logs in the artifact-repo *and* the annotation on the pod. and then this loop will only call a single func.
or wrap `SaveLogs` and `AnnotateOutputs` in a new func. whichever you prefer.

Reply:
nvm. i realized `SaveLogs` shouldn't contain `AnnotateOutputs`. the 2nd might happen without the 1st.
i think i got it now in the correct place - at the end of the `waitContainer` call, saving the logs if needed, and afterwards (in any case) save the output to the pod's annotation (using the containerName)